### ISSUE TITLE
Error: "Do not use constructor injection" when opening a project

### ISSUE THREAD

**Issue by gushill-ls** [flutter/flutter-intellij#6613]
## What happened

Opened my project

## Version information

IntelliJ IDEA `2023.1 EAP` • Flutter plugin `io.flutter 72.0.5` • Dart plugin `231.6890.12`

Flutter 3.7.1 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 7048ed95a5 (8 days ago) • 2023-02-01 09:07:31 -0800
Engine • revision 800594f1f4
Tools • Dart 2.19.1 • DevTools 2.20.1

## Exception

null

```
java.lang.Throwable: Do not use constructor injection (requestorClass=io.flutter.editor.ActiveEditorsOutlineService)
	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:202)
	at com.intellij.serviceContainer.ConstructorInjectionKt.instantiateUsingPicoContainer(constructorInjection.kt:78)
	at com.intellij.serviceContainer.ComponentManagerImpl.instantiateClassWithConstructorInjection(ComponentManagerImpl.kt:1029)
	at com.intellij.serviceContainer.ServiceComponentAdapter.createAndInitialize(ServiceComponentAdapter.kt:43)
	at com.intellij.serviceContainer.ServiceComponentAdapter.doCreateInstance(ServiceComponentAdapter.kt:39)
	at com.intellij.serviceContainer.BaseComponentAdapter.doCreateInstance(BaseComponentAdapter.kt:154)
	at com.intellij.serviceContainer.BaseComponentAdapter.createInstance$lambda$1(BaseComponentAdapter.kt:133)
	at com.intellij.openapi.progress.Cancellation.computeInNonCancelableSection(Cancellation.java:95)
	at com.intellij.serviceContainer.BaseComponentAdapter.createInstance(BaseComponentAdapter.kt:132)
	at com.intellij.serviceContainer.BaseComponentAdapter.getInstance(BaseComponentAdapter.kt:92)
	at com.intellij.serviceContainer.ComponentManagerImpl.doGetService(ComponentManagerImpl.kt:702)
	at com.intellij.serviceContainer.ComponentManagerImpl.getService(ComponentManagerImpl.kt:658)
	at io.flutter.editor.ActiveEditorsOutlineService.getInstance(ActiveEditorsOutlineService.java:63)
	at io.flutter.editor.WidgetIndentsHighlightingPassFactory.<init>(WidgetIndentsHighlightingPassFactory.java:82)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:480)
	at com.intellij.serviceContainer.ConstructorInjectionKt.instantiateUsingPicoContainer(constructorInjection.kt:91)
	at com.intellij.serviceContainer.ComponentManagerImpl.instantiateClassWithConstructorInjection(ComponentManagerImpl.kt:1029)
	at com.intellij.serviceContainer.ServiceComponentAdapter.createAndInitialize(ServiceComponentAdapter.kt:43)
	at com.intellij.serviceContainer.ServiceComponentAdapter.doCreateInstance(ServiceComponentAdapter.kt:39)
	at com.intellij.serviceContainer.BaseComponentAdapter.doCreateInstance(BaseComponentAdapter.kt:154)
	at com.intellij.serviceContainer.BaseComponentAdapter.createInstance$lambda$1(BaseComponentAdapter.kt:133)
	at com.intellij.openapi.progress.Cancellation.computeInNonCancelableSection(Cancellation.java:95)
	at com.intellij.serviceContainer.BaseComponentAdapter.createInstance(BaseComponentAdapter.kt:132)
	at com.intellij.serviceContainer.BaseComponentAdapter.getInstance(BaseComponentAdapter.kt:92)
	at com.intellij.serviceContainer.ComponentManagerImpl.doGetService(ComponentManagerImpl.kt:702)
	at com.intellij.serviceContainer.ComponentManagerImpl.getService(ComponentManagerImpl.kt:658)
	at io.flutter.editor.WidgetIndentsHighlightingPassFactory.getInstance(WidgetIndentsHighlightingPassFactory.java:72)
	at io.flutter.editor.WidgetIndentsHighlightingPassFactoryRegistrar.registerHighlightingPassFactory(WidgetIndentsHighlightingPassFactoryRegistrar.java:26)
	at com.intellij.codeInsight.daemon.impl.TextEditorHighlightingPassRegistrarImpl.lambda$reRegisterFactories$0(TextEditorHighlightingPassRegistrarImpl.java:70)
	at com.intellij.openapi.extensions.impl.ExtensionProcessingHelper.forEachExtensionSafe(ExtensionProcessingHelper.kt:32)
	at com.intellij.openapi.extensions.ExtensionPointName.forEachExtensionSafe(ExtensionPointName.kt:49)
	at com.intellij.codeInsight.daemon.impl.TextEditorHighlightingPassRegistrarImpl.reRegisterFactories(TextEditorHighlightingPassRegistrarImpl.java:70)
	at com.intellij.codeInsight.daemon.impl.TextEditorHighlightingPassRegistrarImpl.<init>(TextEditorHighlightingPassRegistrarImpl.java:46)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:480)
	at com.intellij.serviceContainer.ConstructorInjectionKt.instantiateUsingPicoContainer(constructorInjection.kt:91)
	at com.intellij.serviceContainer.ComponentManagerImpl.instantiateClassWithConstructorInjection(ComponentManagerImpl.kt:1029)
	at com.intellij.serviceContainer.ServiceComponentAdapter.createAndInitialize(ServiceComponentAdapter.kt:43)
	at com.intellij.serviceContainer.ServiceComponentAdapter.doCreateInstance(ServiceComponentAdapter.kt:39)
	at com.intellij.serviceContainer.BaseComponentAdapter.doCreateInstance(BaseComponentAdapter.kt:154)
	at com.intellij.serviceContainer.BaseComponentAdapter.createInstance$lambda$1(BaseComponentAdapter.kt:133)
	at com.intellij.openapi.progress.Cancellation.computeInNonCancelableSection(Cancellation.java:99)
	at com.intellij.serviceContainer.BaseComponentAdapter.createInstance(BaseComponentAdapter.kt:132)
	at com.intellij.serviceContainer.BaseComponentAdapter.getInstance(BaseComponentAdapter.kt:92)
	at com.intellij.serviceContainer.ComponentManagerImpl.doGetService(ComponentManagerImpl.kt:702)
	at com.intellij.serviceContainer.ComponentManagerImpl.getService(ComponentManagerImpl.kt:658)
	at com.intellij.codeHighlighting.TextEditorHighlightingPassRegistrar.getInstance(TextEditorHighlightingPassRegistrar.java:15)
	at com.intellij.codeInsight.daemon.impl.TextEditorHighlightingPassRegistrarEx.getInstanceEx(TextEditorHighlightingPassRegistrarEx.java:34)
	at com.intellij.codeInsight.daemon.impl.TextEditorBackgroundHighlighter.getPasses(TextEditorBackgroundHighlighter.java:74)
	at com.intellij.codeInsight.daemon.impl.DaemonCodeAnalyzerImpl.lambda$submitInBackground$23(DaemonCodeAnalyzerImpl.java:1081)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:889)
	at com.intellij.openapi.application.ReadAction.compute(ReadAction.java:69)
	at com.intellij.codeInsight.daemon.impl.DaemonCodeAnalyzerImpl.lambda$submitInBackground$24(DaemonCodeAnalyzerImpl.java:1070)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$13(CoreProgressManager.java:604)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:679)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:635)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:603)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:60)
	at com.intellij.codeInsight.daemon.impl.DaemonCodeAnalyzerImpl.submitInBackground(DaemonCodeAnalyzerImpl.java:1067)
	at com.intellij.codeInsight.daemon.impl.DaemonCodeAnalyzerImpl.lambda$queuePassesCreation$19(DaemonCodeAnalyzerImpl.java:1032)
	at com.intellij.concurrency.JobLauncherImpl$VoidForkJoinTask$1.exec(JobLauncherImpl.java:181)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
```

**Comment by stevemessick** [comment]
After updating to the latest EAP, I'm seeing this, too. AFAIK we are using standard platform services and should not be getting this. I'll file a bug with JetBrains.
https://youtrack.jetbrains.com/issue/WEB-59477/Do-not-use-constructor-injection-when-Flutter-plugin-starts

**Comment by alexander-doroshko** [comment]
@stevemessick  The error means that the project services shouldn't cause other services creation at the time of initialization. As a rule, services shouldn't keep references to other services as fields. Just use `OtherService.getInstance()` when you need it.

So, the fix is to get rid of `analysisServer` field in the `ActiveEditorsOutlineService` class and use `FlutterDartAnalysisServer.getInstance(project)` directly when needed. You may need to review other services classes as well.

**Comment by stevemessick** [comment]
Thanks @alexander-doroshko. I may need to do a point release to fix this before EAP is stable.

**Comment by stevemessick** [comment]
@gushill-ls If you want to make that error go away, install (from disk) the plugin in the attached file. IntelliJ is going to tell you the plugin needs to be updated, but you should ignore that.

[flutter-intellij.zip](https://github.com/flutter/flutter-intellij/files/10712053/flutter-intellij.zip)

**Comment by stevemessick** [comment]
This is fixed in the latest version, 72.1. Please upgrade your Flutter plugin.
